### PR TITLE
Add deck types and replace group types in frontend

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr adapts the frontend to the new group and deck types.

### CODE REVIEW

File: frontend/components/form/InputField.tsx [hunk 1]
```diff
@@ -118,21 +73,19 @@ function getIcon(status: keyof typeof statusIcon, id: string): ReactNode {
...
-			responsive = true,
+			inputFieldStyle = "primary",
+			inputFieldSize = "md",
 			pattern,
 			className = "",
 			onChange = () => {},
 			onBlur = () => {},
 			...props
-		}: InputFieldProps,
+		}: InputFieldProps & InputHTMLAttributes<HTMLInputElement>, //ButtonHTMLAttributes<HTMLButtonElement>,
```
Comment (TomRomeo) [inline frontend/components/form/InputField.tsx:88]: remove comment
```suggestion
		}: InputFieldProps & InputHTMLAttributes<HTMLInputElement>, 
```

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx [hunk 2]
```diff
@@ -69,18 +69,23 @@ export const DeckSettingsTab = ({
...
-							></DangerAction>
-							<hr className="border-kiokuLightBlue" /> */}
+				{/* Settings for group admins */}
+				<DangerAction
+					id="visibilityDeckDangerAction"
+					header="Change deck visibility"
+					description={`This deck is currently ${deck.deckType?.toLowerCase()}.`}
+					button="Change Visibility"
+					disabled={!isAdmin}
+					onClick={() => {
```
Comment (TomRomeo) [inline frontend/components/navigation/Tabs/DeckSettingsTab.tsx:79]: add `onKeyUp` listener to prevent SonarCloud bug.

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 3]
```diff
@@ -87,21 +89,23 @@ export const GroupSettingsTab = ({
...
 				<hr className="border-kiokuLightBlue" />
 				{/* Settings for group admins */}
-				<DangerAction
-					id="visibilityGroupDangerAction"
-					header="Change group visibility"
-					description={`This group is currently ${group.groupType?.toLowerCase()}.`}
-					button="Change Visibility"
+				<ToggleAction
+					id="groupTypeDangerAction"
+					header="Change who can join this group"
```
Comment (TomRomeo) [inline frontend/components/navigation/Tabs/GroupSettingsTab.tsx:94]:
```suggestion
					header="Change how others can join this group"
```
Maybe reword it like this since it isn't really about _who_ can join but _how_ they can join.

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 4]
```diff
@@ -87,21 +89,23 @@ export const GroupSettingsTab = ({
...
+					description={
+						group.groupType === "OPEN"
+							? "Everyone can join this group"
+							: group.groupType === "REQUEST"
+							? "Everyone can request to join this group"
+							: "Nobody can request to join this group"
```
Comment (TomRomeo) [inline frontend/components/navigation/Tabs/GroupSettingsTab.tsx:100]: if changing the above, change these texts too.

File: frontend/components/input/ToggleButtonGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+							? getStyle[activeButtonStyle]
+							: getStyle["inactive"]
+					}`}
+					onClick={(event) => {
+						if (disabled || choice == active) {
+							return;
+						}
+						setActive(event.currentTarget.value);
+						onChange(event);
+					}}
```
Comment (TomRomeo) [inline frontend/components/input/ToggleButtonGroup.tsx:86]: Similarly to above. Please check out if `onKeyUp` handler can be used here in conjunction with tabindex to make the component accessible .

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx (hunk 2 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/DeckSettingsTab.tsx:79]: Not necessary here since the DangerAction component is using a Button.

File: frontend/components/input/ToggleButtonGroup.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/components/input/ToggleButtonGroup.tsx:86]: Like described above this is not necessary here since the component is using a button.

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx (hunk 3 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/GroupSettingsTab.tsx:94]: Great suggestion!

File: frontend/components/form/InputField.tsx (hunk 1 above)
Comment (Julian702) [reply frontend/components/form/InputField.tsx]: Removed in add1ce1a87737ccadc299da594b7024000460400

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx (hunk 4 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/GroupSettingsTab.tsx]: Changed the last one for more consistency in add1ce1a87737ccadc299da594b7024000460400

**Review by TomRomeo — APPROVED** [review]
Great!

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx (hunk 2 above)
Comment (TomRomeo) [reply frontend/components/navigation/Tabs/DeckSettingsTab.tsx:79]: Oh, I didn't catch that!

**Review by memeToasty — APPROVED** [review]

**Review by darmiel — APPROVED** [review]
TA

File: frontend/components/input/ToggleButtonGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	/**
+	 * Unique identifier
+	 */
+	id: string;
+	/**
+	 * List of options that will be displayed as buttons
+	 */
+	choices: string[];
+	/**
+	 *	Initially active button
```
Comment (4KevR) [inline frontend/components/input/ToggleButtonGroup.tsx:15]: Comment format

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 7]
```diff
@@ -87,21 +89,23 @@ export const GroupSettingsTab = ({
...
-					button="Change Visibility"
+				<ToggleAction
+					id="groupTypeDangerAction"
+					header="Change how others join this group"
+					description={
+						group.groupType === "OPEN"
+							? "Everyone can join this group"
+							: group.groupType === "REQUEST"
+							? "Everyone can request to join this group"
+							: "Everyone has to be invited to join this group"
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/GroupSettingsTab.tsx:100]: I think a switch statement could be more readable, but okay if intended

**Review by 4KevR — APPROVED** [review]
